node/internal/parse_yarn_lock.js

```diff
@@ -278,17 +286,18 @@ function printNodeModule(module) {
   print(``);
   printJson(module);
   print(`node_module(`);
-  print(`    name = "${module.name}",`);
+  print(`    name = "${module.yarn ? module.yarn.label : module.name}",`);
 
   // SCC pseudomodule wont have 'yarn' property
   if (module.yarn) {
     const url = module.yarn.url || module.url;
     const sha1 = module.yarn.sha1;
     const executables = module.executables;
-    
+
+    print(`    module_name = "${module.name}",`);
     print(`    version = "${module.version}",`);
     print(`    package_json = "node_modules/${module.name}/package.json",`);
-    print(`    srcs = glob(["node_modules/${module.name}/**/*"], exclude = ["node_modules/${module.name}/package.json"]),`);
+    print(`    srcs = glob(["node_modules/${module.name}/**/*"], exclude = ["node_modules/${module.name}/package.json", "**/* *"]),`);
```

Comment:
Can you comment a bit more on the intent here?  I think it will be confusing to future people looking at it.

Reply:
This one I'm not sure about.

This was actually a crude workaround to deal with the fact that bazel doesn't like filenames with spaces, and some-dep-I-can't-remember in the polymer-cli tree has such a file.

Ironically it looked like it was a module's test data for filenames with spaces, so my initial instinct was to exclude the test/** tree from the imported modules.

But then I realised anything dealing with that should probably be more firmly dealt with (and documented), e.g. the Gradle client-dependencies-plugin which includes/excludes specific stuff from the imported modules. 

```
clientDependencies {
    fileExtensions = ['css', 'js', 'eot', 'svg', 'ttf', 'woff', 'woff2', 'ts',
                      'jpg', 'jpeg', 'png', 'gif'] // <1>
    releaseFolders = ['dist', 'release'] // <2>
    copyIncludes = [] // <3>
    copyExcludes = ['**/*.min.js', '**/*.min.css', '**/*.map', '**/Gruntfile.js',
                        'gulpfile.js', 'source/**'] // <4>
}
```

A solution for this in rules_node would have to allow the user to override this behaviour (both generally and for specific modules), so it gets a bit complicated and probably deserves its own PR.

So yeah, the end result was "just exclude files which have a space in the name" - which gets us past the blocker for polymer-cli, with the caveat that we hope none of our dependencies have such files in their real sources (which would otherwise if included cause bazel to choke anyway).

If you're happy with this approach I'll add another comment to the code.